### ISSUE TITLE
How to prevent VS code from adding brackets after auto completing?

### ISSUE THREAD

**Issue by HyperbolicSpaceman** [Myriad-Dreamin/tinymist#595]
When I use the `ctheorems` package and type `#theo`, vscode autocomplete it to `#theorem()`,  but what I need is `#theorem[]`.

When I use the official web app, it autocomplete it to `#theorem` without brackets, and it still autocomplete other functions like `#bibliography()` with brackets.  I think the autocompletion of the web app is better for me. Anyone know how to modify the settings?

Thanks!

**Comment by Myriad-Dreamin** [comment]
It doesn't support to check content arguments yet. webapp may be doing some magic things inside.

**Comment by Eric-Song-Nop** [comment]
The Typst webapp behaves much more stupid and inconsistent than Tinymist, from my experiment, the webapp will always try to complete the parens for u, the reason why it is not doing so is simply it fails to detect that `#theorem` is indeed a function.

## Examples of the inconsistency:
1. When `#theorem` is defined in same file with the calling:
```typ
// main.typ
#import "@preview/ctheorems:1.1.3": *
#let theorem = thmbox("theorem", "Theorem", fill: rgb("#eeffee"))

// does not know theorem is a function, considered as a constant. (from the completion symbol is a: c I know this)
#theorem
```
2. When `#theorem` is defined in another file but import directly:
```typ
// lib.typ
#import "@preview/ctheorems:1.1.3": *
#let theorem = thmbox("theorem", "Theorem", fill: rgb("#eeffee"))

// main.typ
import "lib.typ": *
// know this is a function, but does not complete the parens
#theorem
```

3. When `#theorem` is defined in another file but imported with module variable:
```typ
// lib.typ
#import "@preview/ctheorems:1.1.3": *
#let theorem = thmbox("theorem", "Theorem", fill: rgb("#eeffee"))

// main.typ
import "lib.typ" as lib
// know this is a function, and complete a paren
#lib.theorem
```

**Comment by Eric-Song-Nop** [comment]
Even though the completion is webapp is inconsistent for customized functions, it does hard coded some functions with some completion patterns: https://github.com/typst/typst/blob/5672cc2a29379f7743f66c46a7192d698a733047/crates/typst-ide/src/complete.rs#L1316

The bracket mode in complete.rs control the behavior

**Comment by Myriad-Dreamin** [comment]
It will be fixed by #848, which completes bracket after functions that accepts an only positional argument having content type. We still don't have finished the type checking, but we have already supported type annotations in docstring to tell lsp to reliably complete brackets at suitable timings.

```typ
/// - body (content): The content of the theorem.
#let theorem(body) = thmbox("theorem", "Theorem", fill: rgb("#eeffee"), body)
```

![image](https://github.com/user-attachments/assets/8794361a-36a6-4e78-a03c-63683b0e0523)

![image](https://github.com/user-attachments/assets/4e0726c0-71e1-4200-9dbb-309b1797decb)
